docs/wiki/ParameterFileTokens.md

```diff
@@ -0,0 +1,172 @@
+# Parameter File Token
+
+This section provides details on the Tokens Replacement Functionality that enables the use of tokens inside template Parameter Files instead of plain text.
+
+---
+
+### _Navigation_
+
+- [Parameter File Token](#parameter-file-token)
+    - [_Navigation_](#navigation)
+  - [Description](#description)
+  - [How it works](#how-it-works)
+    - [Token Types](#token-types)
+      - [1. Default Tokens (Environment Variables) [Default]](#1-default-tokens-environment-variables-default)
+      - [2. Local Custom Tokens (Source Control) [Optional]](#2-local-custom-tokens-source-control-optional)
+      - [3. Remote Custom Tokens (Key Vault) [Optional]](#3-remote-custom-tokens-key-vault-optional)
+    - [Set / Create Azure Key Vault Token Logic](#set--create-azure-key-vault-token-logic)
+    - [Get Parameter File Tokens](#get-parameter-file-tokens)
+
+---
+
+## Description
+
+The Resource Modules Library Pipelines contains a Token replacement function that enables Parameter files to contain tokens (i.e. `<<subscriptionId>>`, `<<tenantId>>`) instead of using static values. This helps with the following:
+
+- Allows the repository to be portable without having static values from where it was cloned.
+- Enables dynamic updates of the tokens from single locations without having to modify all files.
+- Not adding more environment variables to workflows/pipelines whenever new tokens are required for the environment.
+
+## How it works
+
+### Token Types
+
+There are (3) Token types that can be applied on a Parameter File:
+
+#### 1. Default Tokens (Environment Variables) [Default]
+
+These are tokens constructed from Environment Variables, which are defined in the Workflow (Pipeline). Review [Getting Started - GitHub specific prerequisites](./GettingStarted.md##github-specific-prerequisites) for more information on these Environment Variables.
+
+---
+**Note**: Avoid updating tokens as Environment Variables as they do require changes to Workflows in order to add them to the Agent at runtime.
+
+**Note**: Only use secure tokens in a parameter that is of type `secureString` and not regular strings. These tokens will be replaced at runtime and will be available in the deployment data if the Parameter type is not secure. Follow best [practices and guidelines](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#security-recommendations-for-parameters) on how to handle secrets in template deployments.
+
+---
+
+#### 2. Local Custom Tokens (Source Control) [Optional]
+
+These are tokens defined in the Git Repository inside a [Settings.json](../../settings.json) file. This allows creating tokens that are local and updatable via Source Control mechanisms. Here is an example on where these tokens are stored. You can add key value pairs as required:
```

Comment:
Link wont work when this is published to wiki. Link is relative and will not be present on wiki repo.

Reply:
Fixed